sockeye/beam_search.py

```diff
@@ -604,6 +609,10 @@ def forward(self,
         eos_dist = mx.nd.full((batch_size * self.beam_size, self.output_vocab_size), val=np.inf,
                               ctx=self.context, dtype=self.dtype)
         eos_dist[:, C.EOS_ID] = 0
+        unk_dist = None
+        if self.prevent_unk:
+            unk_dist = mx.nd.zeros_like(eos_dist)
```

Comment:
Should we not use the `self.context` here?

Reply:
storage type, context, and data type are taken from the input array of `zeros_like` so `unk_dist` should be on `self.context`.

Reply:
```python
a = mx.nd.ones((1,), ctx=mx.cpu(1))
a2 = mx.nd.ones((1,), ctx=mx.cpu(0))
mx.nd.zeros_like(a).context  # cpu(1)
mx.nd.zeros_like(a2).context  # cpu(0)
```

Reply:
ohh ok, excellent!